### PR TITLE
[cleanup] make module compile with DUNE 2.5

### PR DESCRIPTION
**Pull request by dr-robertk**
This PR provides some minor fixes to make the code compile with the current DUNE release.

### CODE REVIEW

**Comment by dr-robertk** [comment]
jenkins build this with downstreams please

**Comment by blattms** [comment]
Great. Thanks.

**Comment by dr-robertk** [comment]
jenkins build this please

**Comment by dr-robertk** [comment]
@blattms: opm-grid did work as it is in current master. I did not follow the other discussion closely.

**Comment by dr-robertk** [comment]
Testing was disabled that's why it worked. We can continue the discussion in the opm-common thread.

**Comment by dr-robertk** [comment]
jenkins build this with downstreams please

**Comment by dr-robertk** [comment]
jenkins run norne please

**Comment by andlaus** [comment]
jenkins build this with downstreams please

File: ebos/eclpeacemanwell.hh [hunk 1]
```diff
@@ -602,15 +608,15 @@ public:
         DofVariables& dofVars = *dofVariables_[globalDofIdx];
         wellTotalVolume_ += context.model().dofTotalVolume(globalDofIdx);
 
-        dofVars.elementPtr.reset(new ElementPointer(context.element()));
+        dofVars.element.reset( new Element(context.element()) );
```
Comment (andlaus) [inline ebos/eclpeacemanwell.hh]: I guess this should be changed not to be a pointer in the case of Dune >= 2.4, i.e.

```c++
dofVars.element. = context.element());
```

maybe there was a reason for using pointers, though...

File: ewoms/common/start.hh [hunk 2]
```diff
@@ -311,7 +311,7 @@ static inline int start(int argc, char **argv)
             std::cout << e.what() << ". Abort!\n" << std::flush;
         return 1;
     }
-#if ! DUNE_VERSION_NEWER(DUNE_COMMON,3,0)
+#if ! DUNE_VERSION_NEWER(DUNE_COMMON, 2, 5)
```
Comment (andlaus) [inline ewoms/common/start.hh]: Nice catch, `DUNE_VERSION_NEWER(DUNE_COMMON,3,0)` also triggers for Dune 2.4-dev and for Dune 2.5-dev. this is messed up anyway.

**Comment by andlaus** [comment]
jenkins build this with downstreams please

**Comment by andlaus** [comment]
jenkins build this with downstreams please

File: ebos/eclpeacemanwell.hh (hunk 1 above)
Comment (dr-robertk) [inline ebos/eclpeacemanwell.hh]: I have addressed this in the latest commit.

**Comment by dr-robertk** [comment]
jenkins build this with downstreams please

File: ebos/eclpeacemanwell.hh [hunk 3]
```diff
@@ -93,10 +93,11 @@ class EclPeacemanWell : public BaseAuxiliaryModule<TypeTag>
 
     typedef Opm::MathToolbox<Evaluation> Toolbox;
 
-#if ! DUNE_VERSION_NEWER(DUNE_GRID, 2, 4)
-    typedef typename GridView::template Codim<0>::EntityPointer ElementPointer;
-#else
     typedef typename GridView::template Codim<0>::Entity        Element;
+#if DUNE_VERSION_NEWER(DUNE_GRID, 2, 4)
+    typedef Element  ElementStorage;
```
Comment (andlaus) [inline ebos/eclpeacemanwell.hh]: IIRC the reason for this `shared_ptr` mess was that in some Dune versions (<= 2.3?), Elements and/or ElementPointers are not default-constructible. I suppose that this has changed?

**Comment by andlaus** [comment]
jenkins build this with downstreams please

**Comment by andlaus** [comment]
it seems like jenkins does not listen to you in this module for some reason. @akva2: could you please investigate?

**Comment by andlaus** [comment]
jenkins build this with downstreams please

**Comment by andlaus** [comment]
jenkins build this with downstreams please

**Comment by andlaus** [comment]
everybody seems to be fine with this, and I've tested locally that Norne does not regress. Merging -- thanks for the work!

**Pull request merged**